src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -247,6 +194,7 @@ public ResponseEntity<Course> updateCourse(@RequestBody Course updatedCourse) th
         }
 
         // Make sure to preserve associations in updated entity
+        updatedCourse.setId(courseId);
```

Comment:
❓ Question: This feels a little bit strange. Should the updated course not have an ID already?

Reply:
The issue here is that the course could have another ID. An attacker could then update a course they don't have access to.

As fixing the endpoint is out of scope, I decided to override the ID to ensure that the correct course gets updated.